packages/eslint-plugin/src/rules/use-pipe-transform-interface.ts

```diff
@@ -1,45 +1,120 @@
       category: 'Best Practices',
       recommended: 'error',
     },
+    fixable: 'code',
     schema: [],
     messages: {
-      usePipeTransformInterface: `Classes decorated with @${AngularClassDecorators.Pipe} decorator should implement ${PIPE_TRANSFORM} interface`,
+      usePipeTransformInterface: `Pipes should implement ${PIPE_TRANSFORM} interface`,
     },
   },
   defaultOptions: [],
   create(context) {
+    const sourceCode = context.getSourceCode();
+    const selector = `ClassDeclaration[implements=undefined], ClassDeclaration > :matches(TSClassImplements:not([expression.name="${PIPE_TRANSFORM}"], [expression.property.name="${PIPE_TRANSFORM}"]))`;
+
     return {
-      [PIPE_CLASS_DECORATOR](node: TSESTree.Decorator) {
-        const classParent = node.parent as TSESTree.ClassDeclaration;
-        if (getDeclaredInterfaceName(classParent, PIPE_TRANSFORM)) {
-          return;
-        }
+      [selector](node: TSESTree.ClassDeclaration | TSESTree.TSClassImplements) {
+        const {
+          errorNode,
+          implementsNodeReplace,
+          implementsTextReplace,
+        } = getErrorSchemaOptions(node);
 
         context.report({
-          node: classParent,
+          node: errorNode,
           messageId: 'usePipeTransformInterface',
+          fix: (fixer) => [
+            getImportFix(sourceCode, node, fixer),
+            fixer.insertTextAfter(implementsNodeReplace, implementsTextReplace),
+          ],
         });
       },
     };
   },
 });
+
+function getErrorSchemaOptions(
+  node: TSESTree.ClassDeclaration | TSESTree.TSClassImplements,
+) {
+  const [
+    errorNode,
+    implementsNodeReplace,
+    implementsTextReplace,
+  ] = isClassDeclaration(node)
+    ? [node.id!, node.id!, ` implements ${PIPE_TRANSFORM}`]
+    : [
+        (node.parent as TSESTree.ClassDeclaration).id!,
+        node,
+        `, ${PIPE_TRANSFORM}`,
+      ];
+
+  return { errorNode, implementsNodeReplace, implementsTextReplace } as const;
+}
+
+function getImportDeclaration(
+  node: TSESTree.ClassDeclaration | TSESTree.TSClassImplements,
+  module: string,
+): TSESTree.ImportDeclaration | undefined {
+  let parentNode: TSESTree.Node | undefined = node;
+
+  while ((parentNode = parentNode.parent)) {
+    if (parentNode.type !== 'Program') continue;
+
+    return parentNode.body.find(
+      (node) => isImportDeclaration(node) && node.source.value === module,
+    ) as TSESTree.ImportDeclaration | undefined;
+  }
+
+  return parentNode;
+}
+
+function getImportFix(
+  sourceCode: Readonly<SourceCode>,
+  node: TSESTree.ClassDeclaration | TSESTree.TSClassImplements,
+  fixer: RuleFixer,
+) {
+  const importDeclaration = getImportDeclaration(
+    node,
+    ANGULAR_CORE_MODULE_PATH,
+  );
+
+  if (!importDeclaration?.specifiers.length) {
+    return fixer.insertTextAfterRange(
+      [0, 0],
+      `import { ${PIPE_TRANSFORM} } from '${ANGULAR_CORE_MODULE_PATH}';\n`,
+    );
+  }
```

Comment:
This case should not happen as in a pipe file you may already have the import from `@angular/core`, however I prefer to let it here to avoid something strange.